### PR TITLE
[Doc Issue][Adaptive cards: data uri payload size][2427760]

### PR DESCRIPTION
**Pull request by v-sdhakshina**
bot message size limits added

### CODE REVIEW

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/bots/how-to/format-your-bot-messages.md](https://github.com/MicrosoftDocs/msteams-docs/blob/5b130ed41e97f753534f0d5a73074c38b8b3bd08/msteams-platform%2Fbots%2Fhow-to%2Fformat-your-bot-messages.md) | 93 | 3 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/ba2e452d-a71f-4d1a-9261-b6db5abab73f) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/bots/how-to/format-your-bot-messages.md](https://github.com/MicrosoftDocs/msteams-docs/blob/2d6a02dee30a85de70141bea4e08849f9fd4d507/msteams-platform%2Fbots%2Fhow-to%2Fformat-your-bot-messages.md) | 93 | 3 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/befd86fe-b073-4a9d-84ce-ccf243a071b3) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/bots/how-to/format-your-bot-messages.md](https://github.com/MicrosoftDocs/msteams-docs/blob/7e88bc7a3e7e82230961fecca8ae5309d28b7ad7/msteams-platform%2Fbots%2Fhow-to%2Fformat-your-bot-messages.md) | 93 | 3 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/53809e26-8a62-41df-8f85-f10b1993d85d) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

File: msteams-platform/bots/how-to/format-your-bot-messages.md [hunk 1]
```diff
@@ -10,6 +10,9 @@ ms.author: anclear
 
 Message formatting enables you to bring out the best in bot messages. You can format your bot messages to include rich cards that are attachments that contain interactive elements, such as buttons, text, images, audio, video, and so on.
 
+> [!NOTE]
+> The bot message size limit is 50 KB. You can't send bot message if its size exceeds 50 KB.
```
Comment (juna-gupta) [inline msteams-platform/bots/how-to/format-your-bot-messages.md:14]: Please get this reviewed by the bots team.

We shouldn't say "bot message", it's not a type of message.

Comment (juna-gupta) [reply msteams-platform/bots/how-to/format-your-bot-messages.md:14]: And we also list message size limits on another page, - https://docs.microsoft.com/en-us/microsoftteams/limits-specifications-teams

Let's update that too.

Comment (aosolis) [inline msteams-platform/bots/how-to/format-your-bot-messages.md:14]: > You can't send bot message if its size exceeds 50 KB.

Please document what happens when a bot tries to send an activity that exceeds 50KB.

Comment (aosolis) [inline msteams-platform/bots/how-to/format-your-bot-messages.md:14]: This is missing the important info of what counts towards this limit. For example, do data URI encoded images contribute to this limit? Whitespace?
Also, 50KB in what encoding?

Comment (vule-msft) [reply msteams-platform/bots/how-to/format-your-bot-messages.md:14]: Followed up internally on this. Please document the message size limit at 40 KB.

Comment (vule-msft) [reply msteams-platform/bots/how-to/format-your-bot-messages.md:14]: Bot will receive back a status code of 413 (RequestEntityTooLarge) with message containing error code "MessageSizeTooBig"

Comment (vule-msft) [reply msteams-platform/bots/how-to/format-your-bot-messages.md:14]: Limit includes the entire message payload encoded as UTF-16

**Review by vule-msft — CHANGES_REQUESTED** [review]

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/bots/how-to/format-your-bot-messages.md](https://github.com/MicrosoftDocs/msteams-docs/blob/1c276af077954f3dabfc847e06db3c4fa5bfb8f8/msteams-platform%2Fbots%2Fhow-to%2Fformat-your-bot-messages.md) | 94 | 3 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/0eaf1911-5704-4bbc-91ef-453992e57f2c) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/bots/how-to/format-your-bot-messages.md](https://github.com/MicrosoftDocs/msteams-docs/blob/b76e399d0d05c88fb75a77c09665d0fbe3815e6e/msteams-platform%2Fbots%2Fhow-to%2Fformat-your-bot-messages.md) | 94 | 3 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/fefeadeb-8de9-4979-9b01-89802d93ca88) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

**Comment by v-sdhakshina** [comment]
@vule-msft 

Doc updated as per your comments, request you to please review and approve this PR.

File: msteams-platform/bots/how-to/format-your-bot-messages.md [hunk 2]
```diff
@@ -10,6 +10,9 @@ ms.author: anclear
 
 Message formatting enables you to bring out the best in bot messages. You can format your bot messages to include rich cards that are attachments that contain interactive elements, such as buttons, text, images, audio, video, and so on.
 
+> [!NOTE]
+> The bot message size limit is 40 KB. If bot message size limit exceeds 40 KB, bot will receive a status code of `413` (RequestEntityTooLarge) with message containing error code `MessageSizeTooBig`. The bot message size limit includes the entire message payload encoded as UTF-16.
```
Comment (juna-gupta) [inline msteams-platform/bots/how-to/format-your-bot-messages.md:14]: Do we cover base64 encoded images someplace? (that they do not contribute to overall message size?)

**Review by juna-gupta — CHANGES_REQUESTED** [review]

File: msteams-platform/bots/how-to/format-your-bot-messages.md (hunk 2 above)
Comment (vule-msft) [reply msteams-platform/bots/how-to/format-your-bot-messages.md:14]: We can say not including base 64 encoded images in the note above

**Review by vule-msft — CHANGES_REQUESTED** [review]

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/bots/how-to/format-your-bot-messages.md](https://github.com/MicrosoftDocs/msteams-docs/blob/dc416fc101501ea5b8286eb075d5d426cb1a8b6b/msteams-platform%2Fbots%2Fhow-to%2Fformat-your-bot-messages.md) | 94 | 3 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/90a46a76-18d3-4786-acea-a75f184431a1) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

**Review by vule-msft — DISMISSED** [review]

**Review by v-habibaumme — APPROVED** [review]
Approved

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/bots/how-to/format-your-bot-messages.md](https://github.com/MicrosoftDocs/msteams-docs/blob/308e5521b0216540c0a5376b0c4cc0899d4a4a97/msteams-platform%2Fbots%2Fhow-to%2Fformat-your-bot-messages.md) | 98 | 1 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/ed09f479-0779-48a1-97a4-719890536e58) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/bots/how-to/format-your-bot-messages.md](https://github.com/MicrosoftDocs/msteams-docs/blob/a8b411ac97203873e9ae16c58ffd4e88d5fd0905/msteams-platform%2Fbots%2Fhow-to%2Fformat-your-bot-messages.md) | 98 | 1 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/50feb551-8cc6-4775-b09f-b34f746bac3d) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/bots/how-to/format-your-bot-messages.md](https://github.com/MicrosoftDocs/msteams-docs/blob/841e16ab77128ef85345d26322fd23e2528d16de/msteams-platform%2Fbots%2Fhow-to%2Fformat-your-bot-messages.md) | 98 | 1 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/a14bbdf2-6db3-49d2-972b-1d72d8766396) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

**Review by surbhigupta12 — APPROVED** [review]

**Pull request merged**